plugins/embed-optimizer/class-embed-optimizer-tag-visitor.php

```diff
@@ -288,14 +288,35 @@ private function add_preconnect_links( OD_Tag_Visitor_Context $context ): void {
 		$embed_wrapper_xpath = self::get_embed_wrapper_xpath( $processor->get_xpath() );
 
 		foreach ( $this->get_preconnect_urls( $processor ) as $preconnect_url ) {
+			$host = wp_parse_url( $preconnect_url, PHP_URL_HOST );
+
+			// Skip if malformed URL.
+			if ( empty( $host ) ) {
+				continue;
+			}
```

Comment:
I don't think this is necessary. The URLs are a static list and are not variable. The host will always be defined.